### ISSUE TITLE
Relation Controller new forced readOnly option does not propagate to manage form widget 

### ISSUE THREAD

**Issue by gabsource** [octobercms/october#2294]
##### Expected behavior

When forcing the _readOnly_ mode like in the code below, the popup that opens when clicking on a row should be in preview mode.

```
<?= $this->relationRender('my_relation', [ 'readOnly' => $this->vars['formContext'] === 'preview' ]) ?>
```
##### Actual behavior

When forcing _readOnly_ mode, the toolbar, checkboxes are removed but the action on the row is still the _update_ context where the user can modify the data.
It works fine when setting `readOnly: true` in the relation Yaml, but this does not allow to set only readOnly for the preview context of the parent page.
##### Reproduce steps

See the code snippet above
##### October build

develop 61ac68fd5e9f84d16a435f4f7b678195538151d9

**Comment by LukeTowers** [comment]
This is what I encountered when I was attempting to implement something like this myself earlier. I ended up proposing adding an event to the relation controller in order to dynamically modify the relation controller config, perhaps that pull request needs to be revisited @daftspunk octobercms/october#2245

**Comment by gabsource** [comment]
Maybe it just requires to add the `readOnly` options to the `$.oc.relationBehavior.clickViewListRecord.clickViewListRecord` javascript action, so the RelationController can rebind the `readOnly` option in the AJAX callback.

Everything is already in place to handle it with the latest commit to RelationController.
This seems to work : 

In `october.relation.js`
```
        this.clickViewListRecord = function(recordId, relationField, sessionKey, readOnly) {
            var newPopup = $('<a />')

            newPopup.popup({
                handler: 'onRelationClickViewList',
                size: 'huge',
                extraData: {
                    'manage_id': recordId,
                    '_relation_field': relationField,
                    '_session_key': sessionKey,
                    '_relation_read_only': readOnly ? 1 : 0
                }
            })
        }
```

In `RelationController` : 
```
             $defaultOnClick = sprintf(
                "$.oc.relationBehavior.clickViewListRecord(':%s', '%s', '%s', %s)",
                $this->relationModel->getKeyName(),
                $this->field,
                $this->relationGetSessionKey(),
                $this->readOnly ? 1 : 0
            );
```

**Comment by daftspunk** [comment]
@gabsource, correct please submit a PR for it.

**Comment by LukeTowers** [comment]
This has been resolved by #2295. Thanks to @gabsource!
